src/common.c

```diff
@@ -1427,31 +1427,30 @@ uint32_t ofi_bsock_async_done(const struct fi_provider *prov,
 	    (cmsg->cmsg_level != SOL_IPV6 && cmsg->cmsg_type != IPV6_RECVERR)) {
 		FI_WARN(prov, FI_LOG_EP_DATA,
 			"Unexpected cmsg level (!IP) or type (!RECVERR)\n");
-		goto disable;
+		return -EINVAL;
 	}
 
 	serr = (void *) CMSG_DATA(cmsg);
 	if ((serr->ee_origin != SO_EE_ORIGIN_ZEROCOPY) || serr->ee_errno) {
 		FI_WARN(prov, FI_LOG_EP_DATA,
 			"Unexpected sock err origin or errno\n");
-		goto disable;
+		return -EINVAL;
```

Comment:
nit: Can you convert this and the above to -FI_EINVAL, just for consistency.

Reply:
Thanks for the fast review! Sure, I'll fix the error codes.

Reply:
Done